datanode/sqlstore/epochs.go

```diff
@@ -80,10 +80,10 @@ func (es *Epochs) Get(ctx context.Context, ID uint64) (entities.Epoch, error) {
 
 func (es *Epochs) GetByBlock(ctx context.Context, height uint64) (entities.Epoch, error) {
 	defer metrics.StartSQLQuery("Epochs", "GetByBlock")()
-	query := `WITH epochs_current AS (SELECT DISTINCT ON (id) * FROM epochs ORDER BY id, vega_time DESC)
-		SELECT e.id, e.start_time, e.expire_time, e.end_time, e.tx_hash, e.vega_time, bs.height first_block, be.height last_block FROM epochs_current AS e
+	query := `SELECT e.id, e.start_time, e.expire_time, e.end_time, e.tx_hash, e.vega_time, bs.height first_block, be.height last_block FROM epochs AS e
```

Comment:
What was broken about it? Because there is a unittest for this call which was passing and is still passing. Also I think as it stands if you query a block from the current in-progress epoch it'll still return `no resource corresponding to this id`.

Reply:
that was an oversight. I tested it with some block on the database and incidentally it was a block for an epoch that has already finished. The unit test was making the same mistake so it didn't fail. Fixed both, thanks.